lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1191 @@
+
+    def camelize(complex_type):
+        if complex_type is None:
+            return
+        new_type = type(complex_type)()
+        if isinstance(complex_type, dict):
+            for key in complex_type:
+                new_type[camel(key)] = camelize(complex_type[key])
+        elif isinstance(complex_type, list):
+            for i in range(len(complex_type)):
+                new_type.append(camelize(complex_type[i]))
+        else:
+            return complex_type
+        return new_type
+
+    return camelize(snake_dict)
+
+
+def prepare_params_for_modify(module, connection, instance):
+    """extract parameters from module and convert to format for modify_db_instance call
+
+    Select those parameters we want, convert them to AWS CamelCase, change a few from
+    the naming used in the create call to the naming used in the modify call.
+    """
+    before_facts = instance_to_facts(instance)
+    abort_on_impossible_changes(module, before_facts)
+
+    params = prepare_changes_for_modify(module, connection, before_facts)
+    if len(params) == 0:
+        return None
+    params.update(prepare_call_settings_for_modify(module, before_facts))
+
+    return params
+
+
+def prepare_changes_for_modify(module, connection, before_facts):
+    """
+    Select those parameters which are interesting and which we want to change.
+    """
+    force_password_update = module.params.get('force_password_update')
+    will_change = instance_facts_diff(connection, before_facts, module.params)
+    if not will_change:
+        return {}
+    facts_to_change = will_change['after']
+
+    # modify can't handle tags
+    try:
+        del(facts_to_change['tags'])
+    except KeyError:
+        pass
+
+    # convert from fact format to the AWS call CamelCase format.
+    params = snake_dict_to_cap_camel_dict(facts_to_change)
+
+    if facts_to_change.get('db_security_groups'):
+        params['DBSecurityGroups'] = facts_to_change.get('db_security_groups').split(',')
+
+    # modify_db_instance takes DBPortNumber in contrast to
+    # create_db_instance which takes Port
+    try:
+        params['DBPortNumber'] = params.pop('Port')
+    except KeyError:
+        pass
+
+    # You can specify 9.6 when creating a DB and get 9.6.2
+    # We should ignore version if the requested version is
+    # a prefix of the current version
+    engine_version = will_change['before'].get('engine_version')
+    after_engine_version = will_change['after'].get('engine_version')
+    if engine_version is not None and engine_version.startwith(after_engine_version):
+        del(facts_to_change['engine_version'])
+
+    # modify_db_instance does not cope with DBSubnetGroup not moving VPC!
+    try:
+        if before_facts['db_subnet_group_name'] == params.get('DBSubnetGroupName'):
+            del(params['DBSubnetGroupName'])
+    except KeyError:
+        pass
+    if not force_password_update:
+        try:
```

Comment:
For optional-deletes, I like `params.pop('ParamNameWhatever', None` instead of catching `KeyError`